jetty-core/jetty-core-server/src/test/java/org/eclipse/jetty/core/server/handler/HelloHandler.java

```diff
@@ -52,12 +52,15 @@ public String getMessage()
     }
 
     @Override
-    public void handle(Request request) throws Exception
+    public void offer(Request originalRequest, Acceptor acceptor) throws Exception
     {
-        Response response = request.accept();
-        response.setStatus(200);
-        response.setContentType(MimeTypes.Type.TEXT_PLAIN_UTF_8.asString());
-        response.setContentLength(_byteBuffer.remaining());
-        response.write(true, response.getCallback(), _byteBuffer.slice());
+        acceptor.accept(originalRequest, exchange ->
+        {
+            Response response = exchange.getResponse();
+            response.setStatus(200);
+            response.setContentType(MimeTypes.Type.TEXT_PLAIN_UTF_8.asString());
+            response.setContentLength(_byteBuffer.remaining());
+            response.write(true, exchange, _byteBuffer.slice());
+        });
```

Comment:
The simple case is not tooooo ugly.  But there is definitely a danger of a user referencing `originalRequest` in the lambda rather than `exchange.getRequest()` and thus accidentally making it a capturing lambda.